introduction.tex

```diff
@@ -33,12 +33,17 @@ \chapter*{Introduction}
 
 \subsection*{Type theory}
 
-Type theory was originally invented by Bertrand Russell \cite{Russell:1908},\index{Russell, Bertrand} as a device for blocking the paradoxes in the logical foundations of mathematics  that were under investigation at the time. It was later developed as a rigorous formal system in its own right (under the name ``$\lambda$-calculus'') by Alonzo Church \cite{Church:1933cl,Church:1940tu,Church:1941tc}.  Although it is not generally regarded as the foundation for classical mathematics, set theory being more customary, type theory still has numerous applications, especially in computer science and the theory of programming languages~\cite{Pierce-TAPL}.
+The concept of a type as the domain of significance of a propositional function was introduced by Bertrand Russell \cite{Russell:1908},\index{Russell, Bertrand} as a device for blocking the paradoxes in the logical foundations of mathematics  that were under investigation at the time. The next milestone in the development of type theory is the paper by Alonzo Church \cite{Church:1940tu} that connected types with the formal system called $\lambda$-calculus  that Church has developed for dealing with functions. 
```

Comment:
The end of this paragraph, with "Church has developed," could do with some grammatical correction.